### PR TITLE
SALTO-1308 Make Read-Only Field Annotations Hidden

### PR DESCRIPTION
**Pull request by omrifarkash**
_Release Notes_: 
ReadOnly annotations on Salesforce, such as "creatable", "updateable" and "queryable" will not appear on nacl files.

** will change existing workspaces on first fetch **

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/utils.ts [hunk 1]
```diff
@@ -89,12 +89,12 @@ void => {
     log.debug(`added API_NAME=${fullApiName} to ${elem.elemID.name}`)
   }
   if (!isField(elem) && !elem.annotationRefTypes[API_NAME]) {
-    elem.annotationRefTypes[API_NAME] = createRefToElmWithValue(BuiltinTypes.SERVICE_ID)
+    elem.annotationRefTypes[API_NAME] = createRefToElmWithValue(Types.primitiveDataTypes.ServiceId)
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/utils.ts:92]: This is redundant for this ticket and I guess wrong. It should stay `BuiltinTypes.SERVICE_ID`. This is the type of the annotationType api_name. That's not what this ticket is about, which is the Fields' types.

File: packages/cli/e2e_test/salto.test.ts [hunk 2]
```diff
@@ -293,8 +296,8 @@ describe('cli e2e', () => {
         SALESFORCE,
         newObjectElemName,
         {
-          [API_NAME]: BuiltinTypes.SERVICE_ID,
-          [METADATA_TYPE]: BuiltinTypes.SERVICE_ID,
+          [API_NAME]: Types.primitiveDataTypes.ServiceId,
```
Comment (adamro) [inline packages/cli/e2e_test/salto.test.ts:299]: The changes in the e2e test seem irrelevant to the ticket

File: packages/salesforce-adapter/src/filters/workflow.ts [hunk 3]
```diff
@@ -125,7 +126,9 @@ const createDummyWorkflowInstance = async (
   const workflowType = new ObjectType({
     elemID: new ElemID(SALESFORCE, WORKFLOW_METADATA_TYPE),
     fields: {
-      [INSTANCE_FULL_NAME_FIELD]: { refType: createRefToElmWithValue(BuiltinTypes.SERVICE_ID) },
+      [INSTANCE_FULL_NAME_FIELD]: {
+        refType: createRefToElmWithValue(Types.primitiveDataTypes.ServiceId),
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/workflow.ts:130]: This seems also not right to do. This field doesn't need the CustomField's annotations.

File: packages/salesforce-adapter/src/transformers/transformer.ts [hunk 4]
```diff
@@ -1302,7 +1310,7 @@ export type MetadataTypeAnnotations = {
 }
 
 export const metadataAnnotationTypes: Record<keyof MetadataTypeAnnotations, ReferenceExpression> = {
-  [METADATA_TYPE]: createRefToElmWithValue(BuiltinTypes.SERVICE_ID),
+  [METADATA_TYPE]: createRefToElmWithValue(Types.primitiveDataTypes.ServiceId),
```
Comment (adamro) [inline packages/salesforce-adapter/src/transformers/transformer.ts:1313]: redundant

File: packages/salesforce-adapter/src/transformers/transformer.ts [hunk 5]
```diff
@@ -999,7 +1007,7 @@ export const toCustomProperties = async (
 export const getValueTypeFieldElement = (parent: ObjectType, field: ValueTypeField,
   knownTypes: Map<string, TypeElement>, additionalAnnotations?: Values): Field => {
   const naclFieldType = (field.name === INSTANCE_FULL_NAME_FIELD)
-    ? BuiltinTypes.SERVICE_ID
+    ? Types.primitiveDataTypes.ServiceId
```
Comment (adamro) [inline packages/salesforce-adapter/src/transformers/transformer.ts:1010]: Seems redundant as well

**Review by adamro — COMMENTED** [review]
It looks like you changes too many things to be the Salesforce ServiceId instead of the builtinType.

The change should have been only for CustomFields that received the type serviceId since the field type was 'id' (line 1156 in transformer). The rest are not relevant to what the ticket wants to achieve and probably a wrong change.

File: packages/salesforce-adapter/src/constants.ts [hunk 6]
```diff
@@ -59,6 +59,7 @@ export enum FIELD_TYPE_NAMES {
 export enum INTERNAL_FIELD_TYPE_NAMES {
   UNKNOWN = 'Unknown', // internal-only placeholder for fields whose type is unknown
   ANY = 'AnyType',
+  SERVICE_ID = 'ServiceId',
```
Comment (adamro) [inline packages/salesforce-adapter/src/constants.ts:62]: where's this used?

File: packages/salesforce-adapter/test/filters/topics_for_objects.test.ts [hunk 7]
```diff
@@ -50,7 +50,7 @@ describe('Topics for objects filter', () => {
       [ENABLE_TOPICS]: { refType: createRefToElmWithValue(BuiltinTypes.BOOLEAN) },
       [ENTITY_API_NAME]: { refType: createRefToElmWithValue(BuiltinTypes.STRING) },
       [constants.INSTANCE_FULL_NAME_FIELD]: {
-        refType: createRefToElmWithValue(BuiltinTypes.SERVICE_ID),
+        refType: createRefToElmWithValue(Types.primitiveDataTypes.ServiceId),
```
Comment (adamro) [inline packages/salesforce-adapter/test/filters/topics_for_objects.test.ts:53]: this also seems unrelated

File: packages/salesforce-adapter/test/filters/extra_dependencies.test.ts [hunk 8]
```diff
@@ -50,7 +50,9 @@ describe('extra dependencies filter', () => {
       {
         fields: {
           fieldName: { refType: createRefToElmWithValue(BuiltinTypes.STRING) },
-          [INSTANCE_FULL_NAME_FIELD]: { refType: createRefToElmWithValue(BuiltinTypes.SERVICE_ID) },
+          [INSTANCE_FULL_NAME_FIELD]: {
+            refType: createRefToElmWithValue(Types.primitiveDataTypes.ServiceId),
```
Comment (adamro) [inline packages/salesforce-adapter/test/filters/extra_dependencies.test.ts:54]: also seems unrelated

File: packages/salesforce-adapter/test/mock_elements.ts [hunk 9]
```diff
@@ -34,7 +34,7 @@ const createMetadataObjectType = (
   ...params,
   fields: {
     [INSTANCE_FULL_NAME_FIELD]: {
-      refType: createRefToElmWithValue(BuiltinTypes.SERVICE_ID),
+      refType: createRefToElmWithValue(Types.primitiveDataTypes.ServiceId),
```
Comment (adamro) [inline packages/salesforce-adapter/test/mock_elements.ts:37]: also seems unrelated

File: packages/salesforce-adapter/test/adapter.crud.test.ts [hunk 10]
```diff
@@ -641,7 +641,7 @@ describe('SalesforceAdapter CRUD', () => {
               token: { refType: createRefToElmWithValue(BuiltinTypes.STRING) },
               sandbox: { refType: createRefToElmWithValue(BuiltinTypes.BOOLEAN) },
               [constants.INSTANCE_FULL_NAME_FIELD]: {
-                refType: createRefToElmWithValue(BuiltinTypes.SERVICE_ID),
+                refType: createRefToElmWithValue(Types.primitiveDataTypes.ServiceId),
```
Comment (adamro) [inline packages/salesforce-adapter/test/adapter.crud.test.ts:644]: only relevant for CustomObjects

File: packages/salesforce-adapter/src/constants.ts (hunk 6 above)
Comment (omrifarkash) [reply packages/salesforce-adapter/src/constants.ts:62]: this is used since in transformer `primitiveDataTypes: Record<ALL_FIELD_TYPE_NAMES, PrimitiveType>`  so the keys in this record must be defined in `ALL_FIELD_TYP_NAMES` which is the union of `FIELD_TYPE_NAMES` and `INTERNAL_FIELD_TYPE_NAMES`

File: packages/salesforce-adapter/test/mock_elements.ts [hunk 11]
```diff
@@ -14,7 +14,7 @@
 * limitations under the License.
 */
 import _ from 'lodash'
-import { ObjectType, ElemID, TypeElement, BuiltinTypes, ListType } from '@salto-io/adapter-api'
+import { ObjectType, ElemID, TypeElement, ListType, BuiltinTypes } from '@salto-io/adapter-api'
```
Comment (adamro) [inline packages/salesforce-adapter/test/mock_elements.ts:17]: undo this

File: packages/salesforce-adapter/test/filters/extra_dependencies.test.ts [hunk 12]
```diff
@@ -50,7 +50,9 @@ describe('extra dependencies filter', () => {
       {
         fields: {
           fieldName: { refType: createRefToElmWithValue(BuiltinTypes.STRING) },
-          [INSTANCE_FULL_NAME_FIELD]: { refType: createRefToElmWithValue(BuiltinTypes.SERVICE_ID) },
+          [INSTANCE_FULL_NAME_FIELD]: {
```
Comment (adamro) [inline packages/salesforce-adapter/test/filters/extra_dependencies.test.ts:53]: undo

**Review by adamro — APPROVED** [review]
